lighthouse-core/test/audits/dobetterweb/charset-test.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const CharsetDefinedAudit = require('../../../audits/dobetterweb/charset.js');
+const assert = require('assert');
+const networkRecordsToDevtoolsLog = require('../../network-records-to-devtools-log.js');
+
+/* eslint-env jest */
+
+describe('Charset defined audit', () => {
+  it('succeeds when the page contains the charset meta tag', () => {
+    const finalUrl = 'https://example.com/';
```

Comment:
we try to strike a weird balance between DRY and WET in our tests.

in this case i think this test would benefit from a `getArtifacts()` helper method  that all of these cases can use.  that way it'd be a much easier to see at at glance how each of the artifacts differ.

if you search for `artifacts(` in the `lh-core/test/` folder you'll find a few diff tests that use this sort of pattern.